packages/debug/src/browser/debug-session.tsx

```diff
@@ -414,31 +414,31 @@ export class DebugSession implements CompositeTreeElement {
         for (const thread of this.threads) {
             thread.clear();
         }
-        this.updateCurrentThread();
+        this.updateCurrentThread(false);
     }
-    protected clearThread(threadId: number): void {
+    protected clearThread(forceUpdateSession: boolean, threadId: number): void {
         const thread = this._threads.get(threadId);
         if (thread) {
             thread.clear();
         }
-        this.updateCurrentThread();
+        this.updateCurrentThread(forceUpdateSession);
     }
 
-    protected readonly scheduleUpdateThreads = debounce(() => this.updateThreads(undefined), 100);
+    protected readonly scheduleUpdateThreadsWithoutUpdateSession = debounce(() => this.updateThreads(false, undefined), 100);
     protected pendingThreads = Promise.resolve();
-    updateThreads(stoppedDetails: StoppedDetails | undefined): Promise<void> {
+    updateThreads(forceUpdateSession: boolean, stoppedDetails: StoppedDetails | undefined): Promise<void> {
```

Comment:
This is also a breaking change, any downstream application that calls this method will be broken by the changes.
In order to not break the API, you can use optional parameters.

Reply:
I have set a default true to forceUpdateSession